dingo/gw/waveform_generator/waveform_generator.py

```diff
@@ -1158,6 +1165,45 @@ def generate_TD_modes_L0(self, parameters):
 
         return hlms_lal, parameters_gwsignal["inclination"].value
 
+    def generate_TD_modes_L0_conditioned_extra_time(self, parameters):
+        """
+        Generate TD modes in the L0 frame.
+
+        Parameters
+        ----------
+        parameters: dict
+            Dictionary of parameters for the waveform.
+            For details see see self.generate_hplus_hcross.
+
+        Returns
+        -------
+        hlm_td: dict
+            Dictionary with (l,m) as keys and the corresponding TD modes in lal format as
+            values.
+        iota: float
+        """
+        # TD approximants that are implemented in L0 frame. Currently tested for:
+        #   52: SEOBNRv4PHM
+
+        parameters_gwsignal = self._convert_parameters(
+            {**parameters, "f_ref": self.f_ref}
+        )
+
+        f_min, new_fstart, textra, original_f_min, fisco = wfg_utils.get_starting_frequency_for_conditioning(parameters_gwsignal)
```

Comment:
Keep to a standard variable name convention, e.g., `new_f_start`, `t_extra`, `f_isco`.